src/circularbuffer.jl

```diff
@@ -0,0 +1,104 @@
+
+# this represents a circular buffer, where new items are pushed to the back of the list, 
+# overwriting values in a circular fashion
+# iterating occurs from the "start" until the "end", and will return "length(buffer)" nodes
+
+type CircularBuffer{T} <: AbstractVector{T}
+  capacity::Int
+  startIdx::Int
+  buffer::Vector{T}
+
+  CircularBuffer{T}(::Type{T}, capacity::Int) = new(capacity, 1, T[])
```

Comment:
There is no need to write the type twice, since this is an inner constructor for a parametrized type. This could just be `CircularBuffer(capacity::Int) = new(capacity, 1, T[])`.